### ISSUE TITLE
error encountered while processing 'crt'

### ISSUE THREAD

**Issue by edsilegxrepo** [haproxy/haproxy#574]
## Output of `haproxy -vv` and `uname -a`

```
HA-Proxy version 2.1.4 2020/04/02 - https://haproxy.org/
Status: stable branch - will stop receiving fixes around Q1 2021.
Known bugs: http://www.haproxy.org/bugs/bugs-2.1.4.html
Build options :
  TARGET  = linux-glibc
  CPU     = native
  CC      = /opt/rh/devtoolset-8/root/usr/bin/gcc
  CFLAGS  =
  OPTIONS = USE_PCRE2=1 USE_PCRE2_JIT=1 USE_THREAD=1 USE_PTHREAD_PSHARED=1 USE_REGPARM=1 USE_LINUX_TPROXY=1 USE_LINUX_SPLICE=1 USE_LIBCRYPT=1 USE_CRYPT_H=1 USE_GETADDRINFO=1 USE_OPENSSL=1 USE_LUA=1 USE_ZLIB=1 USE_TFO=1 USE_SYSTEMD=1

Feature list : +EPOLL -KQUEUE -MY_EPOLL -MY_SPLICE +NETFILTER -PCRE -PCRE_JIT +PCRE2 +PCRE2_JIT +POLL -PRIVATE_CACHE +THREAD +PTHREAD_PSHARED +REGPARM -STATIC_PCRE -STATIC_PCRE2 +TPROXY +LINUX_TPROXY +LINUX_SPLICE +LIBCRYPT +CRYPT_H -VSYSCALL +GETADDRINFO +OPENSSL +LUA +FUTEX +ACCEPT4 -MY_ACCEPT4 +ZLIB -SLZ +CPU_AFFINITY +TFO +NS +DL +RT -DEVICEATLAS -51DEGREES -WURFL +SYSTEMD -OBSOLETE_LINKER +PRCTL +THREAD_DUMP -EVPORTS

Default settings :
  bufsize = 16384, maxrewrite = 1024, maxpollevents = 200

Built with multi-threading support (MAX_THREADS=64, default=2).
Built with OpenSSL version : OpenSSL 1.1.1e FIPS  17 Mar 2020
Running on OpenSSL version : OpenSSL 1.1.1e FIPS  17 Mar 2020
OpenSSL library supports TLS extensions : yes
OpenSSL library supports SNI : yes
OpenSSL library supports : TLSv1.0 TLSv1.1 TLSv1.2 TLSv1.3
Built with Lua version : Lua 5.3.5
Built with network namespace support.
Built with transparent proxy support using: IP_TRANSPARENT IPV6_TRANSPARENT IP_FREEBIND
Built with PCRE2 version : 10.34 2019-11-21
PCRE2 library supports JIT : yes
Encrypted password support via crypt(3): yes
Built with zlib version : 1.2.11
Running on zlib version : 1.2.11
Compression algorithms supported : identity("identity"), deflate("deflate"), raw-deflate("deflate"), gzip("gzip")

Available polling systems :
      epoll : pref=300,  test result OK
       poll : pref=200,  test result OK
     select : pref=150,  test result OK
Total: 3 (3 usable), will use epoll.

Available multiplexer protocols :
(protocols marked as <default> cannot be specified using 'proto' keyword)
              h2 : mode=HTTP       side=FE|BE     mux=H2
            fcgi : mode=HTTP       side=BE        mux=FCGI
       <default> : mode=HTTP       side=FE|BE     mux=H1
       <default> : mode=TCP        side=FE|BE     mux=PASS

Available services : none

Available filters :
        [SPOE] spoe
        [CACHE] cache
        [FCGI] fcgi-app
        [TRACE] trace
        [COMP] compression

Linux gfusw-dmft700 3.10.0-1062.12.1.el7.x86_64 #1 SMP Tue Feb 4 23:02:59 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

## What's the configuration?

```
frontend  mftdev-HTTPS443
  bind :443 ssl crt wccusolab.pem alpn h2,http/1.1
  option httplog
  option forwardfor except 127.0.0.0/8

```

## Steps to reproduce the behavior

1. haproxy -c -f /etc/haproxy/haproxy.cfg
2. 
3. 

## Do you have any idea what may have caused this?
Upgrade to 2.1.4

## Do you have an idea how to solve the issue?
None

**Comment by edsilegxrepo** [comment]
## Actual behavior
```
[ALERT] 096/032117 (835) : parsing [/etc/haproxy/haproxy.cfg:54] : 'bind :443' : error encountered while processing 'crt'.
[ALERT] 096/032117 (835) : Error(s) found in configuration file : /etc/haproxy/haproxy.cfg
[ALERT] 096/032117 (835) : Fatal errors found in configuration.
```

## Expected behavior
No error, as none observed in haproxy 2.0.13

**Comment by wlallemand** [comment]
That's strange, we didn't change anything regarding the certificate loading between 2.1.3 and 2.1.4. Did you change your openssl version? Could you try to run the reg-tests with `make reg-tests` to check if everything works correctly?

**Comment by edsilegxrepo** [comment]
Thank you, please close the case.
The issue was due to the fact that the user running the haproxy -c -f /etc/haproxy.cfg
command did not have read access to the pem file mentioned in the bind statement.

**Comment by wlallemand** [comment]
Thanks for the feedback, I'll keep it open to fix the error output because it's not clear at all and don't help the user.

**Comment by wlallemand** [comment]
I fixed the error output and backported it to 2.1.
